### PR TITLE
Support provhdr on Stratis altcoin

### PR DESCRIPTION
**Pull request by zeptin**
Intended to supersede #692

### CODE REVIEW

**Comment by NicolasDorier** [comment]
rebase?

File: NBitcoin/RPC/RestClient.cs [hunk 1]
```diff
@@ -76,7 +76,7 @@ public RestClient(Uri address, Network network)
 		/// <param name="blockId">The block identifier.</param>
 		/// <returns>Given a block hash (id) returns the requested block object.</returns>
 		/// <exception cref="System.ArgumentNullException">blockId cannot be null.</exception>
-		public async Task<Block> GetBlockAsync(uint256 blockId)
+		public async Task<Block> GetBlockAsync(uint256 blockId, int verbosity = 0)
```
Comment (NicolasDorier) [inline NBitcoin/RPC/RestClient.cs]: should be enum

Comment (zeptin) [reply NBitcoin/RPC/RestClient.cs]: Both `RestClient` and `RPCClient` inherit from `IBlockRepository`, which defines a method `Task<Block> GetBlockAsync(uint256 blockId, int verbosity);`. So I'm not sure what the best approach would be, unless perhaps the enum was renamed and moved to be used by both implementations.

**Comment by zeptin** [comment]
I've got this branch into a mess, reopening as a new PR

**Pull request closed without merging**